components/script/dom/htmlinputelement.rs

```diff
@@ -81,7 +86,7 @@ pub struct HTMLInputElement {
     // https://html.spec.whatwg.org/multipage/#concept-input-value-dirty-flag
     value_dirty: Cell<bool>,
 
-    // TODO: selected files for file input
+    filelist: DOMRefCell<Option<JS<FileList>>>,
```

Comment:
The Option lets us avoid allocating extra JavaScript objects in this specific case. A more holistic/general solution would have some form of reflector lazy loading so that JS isn't hit as much. We have discussed this in the past though there isn't anything concrete yet. But I'd prefer a general solution over a bunch of specific ones.